webapp/channels/src/components/searchable_channel_list.tsx

```diff
@@ -264,33 +259,70 @@ export default class SearchableChannelList extends React.PureComponent<Props, St
             this.props.hideJoinedChannelsPreference(true);
         }
     };
-
-    render() {
-        const channels = this.props.channels;
-        let listContent;
-        let nextButton;
-        let previousButton;
-
-        let emptyStateMessage = (
-            <FormattedMessage
-                id={this.props.shouldShowArchivedChannels ? t('more_channels.noArchived') : t('more_channels.noPublic')}
-                tagName='strong'
-                defaultMessage={this.props.shouldShowArchivedChannels ? 'No archived channels' : 'No public channels'}
-            />
-        );
-
+    getEmptyStateMessage = () => {
         if (this.state.channelSearchValue.length > 0) {
-            emptyStateMessage = (
+            return (
                 <FormattedMessage
                     id='more_channels.noMore'
                     tagName='strong'
                     defaultMessage='No results for {text}'
-                    values={{
-                        text: this.state.channelSearchValue,
-                    }}
+                    values={{text: this.state.channelSearchValue}}
+                />
+            );
+        }
+        switch (this.props.filter) {
+        case 'archived':
+            return (
+                <FormattedMessage
+                    id={t('more_channels.noArchived')}
+                    tagName='strong'
+                    defaultMessage={'No archived channels'}
+                />
+            );
+        case 'private':
+            return (
+                <FormattedMessage
+                    id={t('more_channels.noPrivate')}
+                    tagName='strong'
+                    defaultMessage={'No private channels'}
+                />
+            );
+        case 'public':
+            return (
+                <FormattedMessage
+                    id={t('more_channels.noPublic')}
+                    tagName='strong'
+                    defaultMessage={'No public channels'}
+                />
+            );
+        default:
+            return (
+                <FormattedMessage
+                    id={t('more_channels.noChannels')}
+                    tagName='strong'
+                    defaultMessage={'No channels'}
                 />
             );
         }
+    };
+    getFilterLabel = () => {
+        switch (this.props.filter) {
+        case 'archived':
+            return <span>{localizeMessage('more_channels.show_archived_channels', 'Channel Type: Archived')}</span>;
+        case 'public':
+            return <span>{localizeMessage('more_channels.show_public_channels', 'Channel Type: Public')}</span>;
+        case 'private':
+            return <span>{localizeMessage('more_channels.show_private_channels', 'Channel Type: Private')}</span>;
+        default:
+            return <span>{localizeMessage('more_channels.show_all_channels', 'Channel Type: All')}</span>;
```

Comment:
if you want the span you can just write <FormattedMessage/> , span are included in this.

Reply:
Sorry @M-ZubairAhmed it is not clear to me where is span included?

Reply:
Good point, updated as suggested